### PR TITLE
cleanup chai usage from remaining unit tests

### PR DESCRIPTION
**Pull request by say25**
## Overview

Speeding up the migration from `mocha` & `chai` to `jest`.

## Description

- Removing Chai usage in TypeScript Tests.

## Release notes

Notes: No notes.

### CODE REVIEW

**Comment by shiftkey** [comment]
@say25 I started bashing my head against these tests once I realised the issue yesterday, but NYE meant I didn't update the PR to add this context

Thinking on this some more, we can do this transition without continuing to stack pull requests:

 - I'll update #6518 to not remove the `mocha` and `chai` packages (should make the build green again)
 - You can open a fresh PR targeting `development` (rather than targeting #6518) to tidy up the remaining `chai` usages in the unit tests

Once both those are in, we can remove the packages from `package.json`.

File: app/test/unit/release-notes-test.ts [hunk 1]
```diff
@@ -12,27 +10,27 @@ describe('release-notes', () => {
 
       const result = parseReleaseEntries(values)
 
-      expect(result.length).to.equal(1)
-      expect(result[0].kind).to.equal('fixed')
-      expect(result[0].message).to.equal('something else')
+      expect(result.length).toBe(1)
```
Comment (shiftkey) [inline app/test/unit/release-notes-test.ts]:
```suggestion
      expect(result).toHaveLength(1)
```

File: app/test/unit/release-notes-test.ts [hunk 2]
```diff
@@ -12,27 +10,27 @@ describe('release-notes', () => {
...
     })
     it('formats uppercased fixed message', () => {
       const values = ['[Fixed] and another thing']
 
       const result = parseReleaseEntries(values)
 
-      expect(result.length).to.equal(1)
-      expect(result[0].kind).to.equal('fixed')
-      expect(result[0].message).to.equal('and another thing')
+      expect(result.length).toBe(1)
```
Comment (shiftkey) [inline app/test/unit/release-notes-test.ts]:
```suggestion
      expect(result).toHaveLength(1)
```

File: app/test/unit/release-notes-test.ts [hunk 3]
```diff
@@ -12,27 +10,27 @@ describe('release-notes', () => {
...
     })
     it('uses [Other] for unrecognized category', () => {
       const values = ['[Foo] we did a thing!']
 
       const result = parseReleaseEntries(values)
 
-      expect(result.length).to.equal(1)
-      expect(result[0].kind).to.equal('other')
-      expect(result[0].message).to.equal('we did a thing!')
+      expect(result.length).toBe(1)
```
Comment (shiftkey) [inline app/test/unit/release-notes-test.ts]:
```suggestion
      expect(result).toHaveLength(1)
```

File: app/test/unit/release-notes-test.ts [hunk 4]
```diff
@@ -67,13 +65,13 @@ describe('release-notes', () => {
...
+      expect(result.latestVersion).toBe('1.0.11')
       // the generated date here is local time, so it might either be the 5th or 6th
       // let's just test it's showing the right month and year instead
-      expect(result.datePublished).contains('December')
-      expect(result.datePublished).contains('2017')
-      expect(result.bugfixes.length).to.equal(8)
-      expect(result.enhancements.length).to.equal(12)
+      expect(result.datePublished).toContain('December')
+      expect(result.datePublished).toContain('2017')
+      expect(result.bugfixes.length).toBe(8)
```
[the remark this answers is not in the document]
Comment (j-f1) [reply app/test/unit/release-notes-test.ts]: This is the kind of thing that snapshots might make much easier.

File: app/test/unit/diff-parser-test.ts [hunk 5]
```diff
@@ -237,35 +235,35 @@ index 1910281..257cc56 100644
...
-    expect(lines[i].text).to.equal('-foo')
-    expect(lines[i].type).to.equal(DiffLineType.Delete)
-    expect(lines[i].oldLineNumber).to.equal(1)
-    expect(lines[i].newLineNumber).to.equal(null)
-    expect(lines[i].noTrailingNewLine).to.be.true
+    expect(lines[i].text).toBe('-foo')
+    expect(lines[i].type).toBe(DiffLineType.Delete)
+    expect(lines[i].oldLineNumber).toBe(1)
+    expect(lines[i].newLineNumber).toBeNull()
+    expect(lines[i].noTrailingNewLine).toBeTruthy()
```
[the remark this answers is not in the document]
Comment (say25) [reply app/test/unit/diff-parser-test.ts]: Done

**Comment by shiftkey** [comment]
@say25 thanks for helping migrate these over!

**Pull request merged**